src/Parser.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    var $path;
+    var $header_array = array();
+    var $delimiter;
+
+    function __construct ($path, $header_array = array(), $delimiter = ",") {
+
+        if(file_exists($path)){ //verify path
+            $this -> $path = $path; //fill path
+
+            if($delimiter == "," || $delimiter == "|" || $delimiter == "\t") { //verify delimiter

```

Comment:
you can simplify the check using the built in function  [in_array()](https://github.com/xavier643/fluent-csv-php-parser.git). You should also return some sort of error if an invalid delimiter was specified rather than allowing the Object to be constructed without a delimiter which could cause abnormal behaviour when trying to parse a CSV File.

Reply:
I'll have to research this seeing as how delimiter is not an array but just a single variable.  I can understand the in_array() working on the header however.